check parser_dates names in columns

sathyz opened the pull request:
if column names are passed in parser_dates, make sure those
columns exist in dataframe.

- [x] closes #31251 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

sathyz committed:
check parser_dates names in columns

if column names are passed in parser_dates, make sure those
columns exist in dataframe.

ShaharNaveh reviewed (changes requested):
Please run "Black" and then push again:)

jbrockmendel committed:
REF: Remove CategoricalIndex.get_value (#31765)

jbrockmendel committed:
CLN: tests.indexing.common (#31812)

jbrockmendel committed:
REF: tighten what we accept in TimedeltaIndex._simple_new (#31315)

jbrockmendel committed:
CLN: misplaced TimedeltaIndex tests (#31755)

jbrockmendel committed:
TST: parametrize some indexing tests (#31767)

jbrockmendel committed:
BUG: iloc setitem with 3d indexer not raising (#31813)

jbrockmendel committed:
BUG: Series[dim3array] failing to raise ValueError for some Index subclasses (#31816)

jbrockmendel committed:
REF: share code for set-like ops in DTI/TDI/PI (#31335)

jbrockmendel committed:
REF: share _partial_date_slice between PeriodIndex/DatetimeIndex (#31766)

jbrockmendel committed:
REF: turn _try_mi into MultiIndex._get_values_for_loc (#31736)

jbrockmendel committed:
REF/TST: misplaced tests in tests.indexes.period (#31758)

jbrockmendel committed:
REF: remove iloc case from _convert_slice_indexer (#31786)

jbrockmendel committed:
DEPR: indexing Series with single-entry list (#31333)

dsaxton committed:
DOC: Fix style guide typos (#31822)

topper-123 committed:
BUG: Too aggressive typing in NDFrame.align (#31788)

rushabh-v committed:
correct the redirections (#31820)

r0cketr1kky committed:
DOC: Removed numeric_only parameter from pd.DataFrame.mad docs (#31641)

Cardosaum committed:
DOC: '10 minutes to pandas' - <TAB> completion section now show a con… (#31818)

topper-123 committed:
CLN: remove kwargs from signature of (Index|MultiIndex).copy (#31669)

jbrockmendel committed:
REF: use public indexers in groupby.ops (#31814)

SaturnFromTitan committed:
fixed mypy errors in mypy-pandas.tests.extension.decimal.test_decimal (#31730)

raisadz committed:
Test messages test period (#31824)

jbrockmendel committed:
REF: simplify PeriodIndex.get_loc (#31598)

jbrockmendel committed:
CLN: trim unreachable indexing code (#31768)

dhirschfeld committed:
Fix bug in calculation of holidays (#31416)

pedroreys committed:
BUG Decode to UTF-8 the dtype string read from a hdf file (#31756)

SaturnFromTitan committed:
fixed mypy errors in pandas.tests.extension.json.test_json (#31836)

charlesdong1991 committed:
ENH: Allow multi values for index and columns in df.pivot (#30928)

jbrockmendel committed:
CLN: Use self.loc for Series __getitem__ with IntervalIndex (#31830)

dsaxton committed:
Don't use caps (#31829)

jbrockmendel committed:
CLN: disallow kind=None in _convert_slice_indexer (#31832)

raisadz committed:
Test messages test integer (#31831)

jbrockmendel committed:
REF: move loc-only validate_read_indexer to Loc (#31834)

ShaharNaveh committed:
DOC: Fixed example section in pandas/core/dtypes/*.py (#31451)

jbrockmendel committed:
CLN: assorted indexing-related cleanups (#31797)

mroeschke committed:
BUG: Fix raw parameter not being respected in groupby.rolling.apply (#31842)

jbrockmendel committed:
REF: implement ExtensionIndex._concat_same_dtype, use for IntervalIndex (#31635)

jchen2186 committed:
Docstring fixes for PR06 errors (#31838)

mroeschke committed:
BUG: Fix rolling.corr with time frequency (#31841)

* BUG: Fix rolling.corr with time frequency

* Add extra tick in whatsnew

* Clarify whatsnew

jbrockmendel committed:
REF: make _setitem_with_indexer iloc-only (#31837)

iamshwin committed:
DOC: Specify use of google cloud storage for CSVs (#31860)

bashtage committed:
REF: Use nonzero in place of argwhere (#31853)

jbrockmendel committed:
REF: make Series/DataFrame _slice always positional (#31854)

Pearcekieser committed:
DOC: Update documentation DataFrame.nsmallest (#31833)

monicaw218 committed:
CLN: f-string formatting (#31868)

jbrockmendel committed:
CLN: tests.generic (#31865)

sathyz committed:
fix lint issue.

sathyz committed:
fix lint issue.

charlesdong1991 committed:
PLT: Color attributes of medianprops etc are lost in df.boxplot and df.plot.boxplot (#31262)

simonjayhawkins committed:
CLN: D300 Use """triple double quotes""" (#31888)

simonjayhawkins committed:
CLN: D201 No blank lines allowed before function docstring (#31889)

simonjayhawkins committed:
CLN: D208 Docstring is over-indented (#31890)

simonjayhawkins committed:
CLN: D204 1 blank line required after class docstring (#31892)

simongibbons committed:
ENH: Support multi row inserts in to_sql when using the sqlite fallback (#30743)

simonjayhawkins committed:
D209 Multi-line docstring closing quotes should be on a separate line (#31891)

jbrockmendel committed:
TST: parametrize eval tests (#31901)

gonemad97 committed:
DOC: Added documentation for ImportError's (#30912)

simonjayhawkins committed:
TYP: partial typing of masked array (#31728)

naterarmstrong committed:
DOC: Fix divmod return values (#31894)

jbrockmendel committed:
CLN: remove odious kludge (#31907)

dsaxton committed:
BUG: Avoid casting Int to object in Categorical.from_codes (#31794)

jorisvandenbossche committed:
BUG: fix StringArray/PandasArray setitem with slice (#31773)

jorisvandenbossche committed:
BUG: fix parquet roundtrip with unsigned integer dtypes (#31918)

jorisvandenbossche committed:
BUG: fix infer_dtype for StringDtype (#31877)

simonjayhawkins committed:
TYP: pandas/core/dtypes/base.py (#31352)

xhochy committed:
BLD: Run flake8 check on Cython files in pre-commit (#30847)

HH-MWB committed:
CLN: Replace Appender and Substitution with simpler doc decorator (#31060)

Commit:
BUG: MultiIndex intersection with sort=False  does not preserve order (#31312)

sathyz commented:
Hi @MomIsBestFriend , I have fixed it. Could you check again. I'm not sure why CI is failing. How do I investigate.

ShaharNaveh committed:
CLN: Some code cleanups (#31792)

simonjayhawkins committed:
D202 No blank lines allowed after function docstring (#31895)

jbrockmendel committed:
ENH: support datetime64, datetime64tz in nanops.mean, nanops.median (#29941)

3vts committed:
CLN: 29547 replace old string formatting 1 (#31914)

jbrockmendel committed:
CLN: implement _getitem_tuple_same_dim (#31911)

MomIsBestFriend commented:
I am restarting the CI, I can't see what's wrong

jbrockmendel committed:
TST: parametrize generic/internals tests (#31900)

3vts committed:
CLN: 29547 replace old string formatting 2 (#31933)

dsaxton committed:
BUG: Handle NA in assert_numpy_array_equal (#31910)

jbrockmendel committed:
CLN: assorted cleanups (#31938)

jorisvandenbossche committed:
TST: expand tests for ExtensionArray setitem with nullable arrays (#31741)

MarcoGorelli committed:
remove blocking return (#31960)

simonjayhawkins committed:
D412: No blank lines allowed between a section header and its content (#31956)

simonjayhawkins committed:
D409: Section underline should match the length of its name (#31958)

simonjayhawkins committed:
D411: Missing blank line before section (#31959)

simonjayhawkins committed:
CLN: D414: Section has no content (#31961)

3vts committed:
CLN: 29547 replace old string formatting 3 (#31945)

3vts committed:
CLN: 29547 replace old string formatting 4 (#31963)

topper-123 committed:
CLN: index related attributes on Series/DataFrame (#31953)

MarcoGorelli committed:
add eval examples (#31955)

jbrockmendel committed:
CLN: remove unreachable in Series._reduce (#31932)

martinbjeldbak committed:
Update documentation to use recommended library (#31968)

3vts committed:
CLN: 29547 replace old string formatting 5 (#31967)

ShaharNaveh committed:
STY: Fixed wrong placement of whitespace (#31974)

3vts committed:
CLN: 29547 replace old string formatting 6 (#31980)

ShaharNaveh committed:
CI: Removed pattern check for specific modules (#31975)

jbrockmendel committed:
TST: parametrize tests.indexing.test_float (#31855)

SaturnFromTitan committed:
started to fixturize pandas/tests/base (#31701)

WillAyd committed:
Revert 31791 (#31931)

alysbrooks committed:
CLN 29547 Replace old string formatting syntax with f-strings (#31982)

za committed:
DOC PR09 Add missing . on freq parameter on groupby.py (#31998)

za committed:
DOC PR09 Add . in the description parameter (#32001)

asyarif93 committed:
DOC PR09 Add missing . on Parameter con description (#32000)

ShaharNaveh committed:
CI: temporary fix to the CI (#32011)

dequadras committed:
DOC: update ohlc docstring so that it reflects the real use #31919 (#31941)

simonjayhawkins committed:
CI: silence numpy-dev failures (#32025)

simonjayhawkins committed:
CLN: D213: Multi-line docstring summary should start at the second line (#31893)

3vts committed:
Seventh batch of changes (#31986)

panjacek committed:
CLN 2574 Replace old string formating (#32007)

pcandoalmeida committed:
CLN: GH29547 replace old string formatting (#32029)

DavaIlhamHaeruzaman committed:
Add period to parameter description (#32005)

za committed:
DOC SS06 Make the summery in one line on offsets.py (#32021)

andhikayusup committed:
DOC: Improve docstring of Index.delete (#32015)

adamwdb committed:
DOC: Update pandas.Series.between_time docstring params (#32014)

DavaIlhamHaeruzaman committed:
DOC: Add missing period to parameter description (#32022)

3vts committed:
Eighth batch of changes (#32032)

simonjayhawkins committed:
CI: change np-dev xfails to not strict (#32031)

WillAyd committed:
Added clang inline helper (#30990)

mroeschke committed:
CLN: Some groupby internals (#31915)

* CLN: Some groupby internals

* Additional annotation

sathyz commented:
All checks passed, could you review and merge.

jreback commented on pandas/io/parsers.py:
```diff
@@ -1423,6 +1424,26 @@ def __init__(self, kwds):
         # keep references to file handles opened by the parser itself
         self.handles = []
 
+    def _confirm_parse_dates_presence(self, columns):
```
rename to _valdate_parse_dates

pls add type annotations for columns

jreback commented on pandas/io/parsers.py:
```diff
@@ -1423,6 +1424,26 @@ def __init__(self, kwds):
+        elif isinstance(self.parse_dates, dict):
+            cols_needed = chain(*self.parse_dates.values())
+        else:
+            cols_needed = []
```
so this is a single scalar column, why is it not needing validation?

sathyz replied:
> parser_dates: bool or list of int or names or list of lists or dict, default False
according to the read_csv doc, this can't be a single scalar.

jreback commented on pandas/io/parsers.py:
```diff
@@ -1423,6 +1424,26 @@ def __init__(self, kwds):
 
+    def _confirm_parse_dates_presence(self, columns):
+        """
+        if user has provided names for parse_dates, check if those columns
```
add a full-docstring (see other validation functions)

jreback commented on pandas/io/parsers.py:
```diff
@@ -1423,6 +1424,26 @@ def __init__(self, kwds):
+        if isinstance(self.parse_dates, list):
+            cols_needed = self.parse_dates
+        elif isinstance(self.parse_dates, dict):
+            cols_needed = chain(*self.parse_dates.values())
```
why do you need the chain here?

gfyoung replied:
This is okay.  The reason is that `parse_dates` as a `dict` maps column names to lists or individual columns.  `chain` is used to "flatten" them out into a single iterable.

jreback commented:
cc @gfyoung

gfyoung commented on pandas/tests/io/parser/test_parse_dates.py:
```diff
@@ -1516,3 +1516,15 @@ def test_hypothesis_delimited_date(date_format, dayfirst, delimiter, test_dateti
+    with pytest.raises(ValueError, match=msg):
+        parser.read_csv(
+            content, sep=",", usecols=["val"], parse_dates=parse_dates,
+        )
```
Add a test where `names` is also passed in

stijnvanhoey committed:
DOC: Update of the 'getting started' pages in the sphinx section of the documentation (#31156)

JustinZhengBC committed:
BUG: list-like to_replace on Categorical.replace is ignored or crash (#31734)

WillAyd committed:
Clean Up C Warnings (#31935)

jbrockmendel committed:
REF: implement unpack_1tuple to clean up Series.__getitem__ (#31906)

jbrockmendel committed:
REF: move loc-only methods to loc (#31859)

dsaxton committed:
CLN: Clean reductions/test_reductions.py (#32035)

TomAugspurger committed:
DOC: pin gitdb2 (#32064)

jbrockmendel committed:
BUG: fix length_of_indexer with boolean mask (#31897)

jorisvandenbossche committed:
WEB: update blog link to only include my pandas blog posts (#32051)

pcandoalmeida committed:
CLN: GH29547 replace old string formatting (#32063)

jbrockmendel committed:
CLN: remove unused from MultiIndex (#32030)

jbrockmendel committed:
REGR: fix op(frame, frame2) with reindex (#31679)

MarcoGorelli committed:
CLN: Move info (#31876)

dsaxton committed:
DOC: Update sort_index docs (#31898)

raisadz committed:
DOC: Mention black and PEP8 in pandas style guide (#32043)

simonjayhawkins committed:
REGR: show_versions (#32041)

simonjayhawkins committed:
TYP: check_untyped_defs core.arrays.categorical (#32097)

raisadz committed:
add messages to tests (#31852)

* add messages to tests

* changes to test_boolean.py

* split error messages

* change to test_isin.py

* changes to test_boolean.py and test_indexing.py

* revert changes to test_diff.py

* cleanups to the code

* changes to test_boolean.py and test_replace.py

* change error message in test_to_dict.py

datapythonista committed:
CLN: Remove unused script find_commits_touching_func.py (#32071)

raisadz committed:
Replace old string formatting syntax with f-strings (#32122)

simonjayhawkins committed:
TYP: check_untyped_defs core.tools.datetimes (#32101)

dsaxton committed:
CLN: Clean groupby/test_function.py (#32027)

WillAyd committed:
Added pd.NA to nulls_fixture (#31799)

jbrockmendel committed:
REF: misplaced Series.combine_first tests (#32111)

amy-graham-js committed:
added msg to TypeError test_to_boolean_array_error (#32103)

mabroor committed:
TST: Fix bare pytest.raises in test_parsing.py (#32102)

simonjayhawkins committed:
TYP: check_untyped_defs arrays.sparse.array (#32099)

jbrockmendel committed:
CLN: simplify _setitem_with_indexer (#31887)

3vts committed:
CLN: 29547 replace old string formatting (#32034)

jbrockmendel committed:
CLN: remove unused tm.isiterable (#32137)

jorisvandenbossche committed:
CI: skip geopandas downstream test (Anaconda installation issue) (#32148)

Dr-Irv committed:
BUG: Fix for convert_dtypes with mix of int and string (#32126)

pedroreys committed:
REGR: read_pickle fallback to encoding=latin_1 upon a UnicodeDecodeError (#32055)

When a reading a pickle with MultiIndex columns generated in py27
`pickle_compat.load()` with `enconding=None` would throw an UnicodeDecodeError
when reading a pickle created in py27. Now, `read_pickle` catches that exception and
fallback to use `latin-1` explicitly.

jbrockmendel committed:
CLN: make tm.N, tm.K private (#32138)

simonjayhawkins committed:
TST: add test for get_loc on tz-aware DatetimeIndex (#32152)

simonjayhawkins committed:
TST: add test for DataFrame.reindex on nearest tz-aware DatetimeIndex (#32155)

jbrockmendel committed:
REG: dont call func on empty input (#32121)

simonjayhawkins committed:
DOC: move whatsnew to sync master with Backport PR #32148 (#32151)

dsaxton committed:
ENH: Enable indexing with nullable Boolean (#31591)

jbrockmendel committed:
REF/TST: collect Index join tests (#32171)

jbrockmendel committed:
REF: organize base class Index tests (#31864)

ShaharNaveh committed:
CLN: some code cleanups in pandas/_libs/ (#31808)

* CLN: some code cleanups in pandas/_libs/

* Reverted "bint"

REF:
https://github.com/pandas-dev/pandas/pull/31808/files#r376721172

* Added trailing comma to imports

REF:
https://github.com/pandas-dev/pandas/pull/31808/files#r378523656

* Reverted bad code

* Lint issues

* Reverted wrong code

REF:
https://github.com/pandas-dev/pandas/pull/31808#discussion_r379806862

* Removed parens

REF:
https://github.com/pandas-dev/pandas/pull/31808#discussion_r381626183

* "in fmt" in prev line

REF:
https://github.com/pandas-dev/pandas/pull/31808#discussion_r381626633

jbrockmendel committed:
BUG: DataFrame.iat incorrectly wrapping datetime objects (#32089)

jbrockmendel committed:
REF/TST: implement test_interpolate for Series (#32112)

jbrockmendel committed:
TST: method-specific files for DataFrame assign, interpolate  (#32110)

jbrockmendel committed:
move method from LocationIndexer to Index (#31857)

jbrockmendel committed:
REF: collect arithmetic benchmarks (#32116)

jbrockmendel committed:
BUG: Fix incorrect _is_scalar_access check in iloc (#32085)

jbrockmendel committed:
PERF: IntegerIndex._shallow_copy (#32130)

jbrockmendel committed:
CLN: NDFrame.__init__ unnecessary code (#32131)

jbrockmendel committed:
CLN: organize MultiIndex indexing tests (#31863)

* implement test_take

* move test_take, organize tests

* cleanup

* remove unused imports

* isort fixup

* typo fixup

jbrockmendel committed:
REF: standardize CategoricalIndex._shallow_copy usage (#32141)

ShaharNaveh committed:
CLN: Some code cleanups (#32176)

jbrockmendel committed:
BUG: disallow invalid dtype to CategoricalDtype._from_values_or_dtype (#32169)

jbrockmendel committed:
REF: de-duplicate object-dtype handling (#32168)

PSY27 committed:
used f-string (#32133)

jbrockmendel committed:
BUG: using loc[int] with object index (#31905)

* BUG: using loc[int] with object index

* whatsnew

jbrockmendel committed:
dont skip keyerror for IntervalIndex (#31936)

SaturnFromTitan committed:
Turned class based tests into function based tests (#32184)

jbrockmendel committed:
CLN: no kwargs for take_with_is_copy (#32181)

jancervenka committed:
CLN: F-string in pandas/tests/indexes/datetimes/test_to_period.py (#29547) (#32189)

dsaxton committed:
BUG: Fix construction of Categorical from pd.NA (#31939)

dsaxton committed:
BUG: Avoid ambiguous condition in GroupBy.first / last (#32124)

MarcoGorelli committed:
BUG: groupby nunique changing values (#32175)

mabroor committed:
TST: Fixed bare pytest.raises in test_window.py (#32158)

jbrockmendel committed:
annotations (#32193)

jbrockmendel committed:
BUG: Index(categorical, dtype=object) not returning object dtype (#32167)

jbrockmendel committed:
CLN: indexing comments and cleanups (#32082)

jbrockmendel committed:
BUG: catch almost-null-slice in _convert_slice_indexer (#31866)

jbrockmendel committed:
TST: parametrize and de-duplicate timedelta64 arithmetic tests (#32091)

jbrockmendel committed:
REF: pass str_rep through arithmetic ops more consistently (#31297)

SaturnFromTitan committed:
Use fixtures in pandas/tests/base (#32046)

jbrockmendel committed:
REF: de-duplicate factorize and duplicated code (#32216)

sumanau7 committed:
Import OptionError in pandas.errors (#32142)

cool-RR committed:
CLN: Fix exception causes in datetimelike.py (#32164)

asyarif93 committed:
DOC: Update pandas.DataFrame.droplevel docstring (#32004)

jbrockmendel committed:
troubleshoot 32bit build (#32241)

SaturnFromTitan committed:
CI: Remove float16 fixture value (#32221)

jbrockmendel committed:
REF/TST: method-specific files for DataFrame timeseries methods (#32230)

jbrockmendel committed:
REF/TST: method-specific files for rename, reset_index (#32227)

jbrockmendel committed:
REF/TST: method-specific files for combine, update; parametrize (#32228)

jbrockmendel committed:
REF/TST: method-specific files for Series timeseries methods (#32226)

dsaxton committed:
CLN/REF: Split up / clean Categorical constructor tests (#32211)

* Split / parametrize

* Dedupe some code

* Parametrize

* Add back from_codes for empty

* Add comment

jbrockmendel committed:
TST: show_versions test with unmerged commits (#32139)

jbrockmendel committed:
implement test_select_dtypes (#32250)

jbrockmendel committed:
test_combine.py (#32253)

jbrockmendel committed:
Collect Series.append tests (#32254)

jbrockmendel committed:
REF: test_to_timestamp (#32248)

jbrockmendel committed:
CLN: generic tests (#32256)

jbrockmendel committed:
PERF: lazify consolidate_check and dont consolidate in get_slice (#32224)

jbrockmendel committed:
CLN: simplify+annotate _shallow_copy (#32244)

jbrockmendel committed:
REF: include CategoricalIndex in index_cached parametrization (#32215)

jbrockmendel committed:
CLN: simplify CategoricalIndex._simple_new (#32204)

fujiaxiang committed:
ENH: Timestamp constructor now raises more explanatory error message (#31653)

dsaxton committed:
ENH: Implement DataFrame.value_counts (#31247)

jbrockmendel committed:
use ExtensionIndex._concat_same_dtype (#32232)

jbrockmendel committed:
REF: simplify IntervalIndex/IntervalArray _shallow_copy (#32247)

jbrockmendel committed:
CLN/TST: parametrize some tests in tests.indexing.test_float (#32187)

* TST: cleanups

* troubleshoot

* troubleshoot travis

schipiga committed:
Add link to pandas-tfrecords (#32143)

jbrockmendel committed:
CLN: remove dtype kwarg from _simple_new (#32260)

AlexKirko committed:
ENH: add fold support to Timestamp constructor (#31563)

jbrockmendel committed:
REGR: preserve freq in DTI/TDI outer join (#32166)

dsaxton committed:
BUG: Cast pd.NA to pd.NaT in to_datetime (#32214)

Dom-L-G committed:
Added message to pytest raises for test_constructor_dict (#32236)

cool-RR committed:
Fix exception causes in 14 modules (#32235)

jbrockmendel committed:
use numexpr for Series comparisons (#32047)

asepwhite committed:
DOC: Fix pandas.index.copy summary documentation (#32006)

laymonage committed:
DOC: Improve documentation for Index.where (#32009)

farhanreynaldo committed:
DOC: Fix errors in pandas.Series.argmax (#32019)

datapythonista committed:
CI: Remove docs build from pipelines (#32074)

raisadz committed:
DOC: add missing links to introduction to pandas (#32198)

jbrockmendel committed:
TST: move misplaced to_datetime test (#32273)

jbrockmendel committed:
Troubleshoot CI (#32284)

jbrockmendel committed:
test_droplevel (#32272)

jaketae committed:
Use defaultdict for optimization (#32278)

jbrockmendel committed:
PERF: pass through to numpy validation for iloc setitem (#32257)

jbrockmendel committed:
TST: misplaced arithmetic tests (#32275)

simonjayhawkins committed:
CLN: Follow-up to #32158 (#32290)

jbrockmendel committed:
TST: implement test_first (#32274)

datapythonista committed:
CI: Temporary fix to the docs build while we fix the ssh problems (#32279)

jbrockmendel committed:
TST: test_to_time (#32285)

jbrockmendel committed:
implement test_to_period (#32270)

mankoff committed:
fixed minor docstring typo (#32298)

jbrockmendel committed:
REF/TST: misplaced tests in test_timeseries, test_timezones (#32300)

Samira-g-js committed:
Added in a error message (#32105)

ryankarlos committed:
DOC: Add example for multiindex series and dataframe merge (#32068)

MarcoGorelli committed:
BUG: fix in categorical merges (#32079)

MarcoGorelli committed:
WEB: Add greeting note to CoC (#32024)

jbrockmendel committed:
API/BUG: raise only KeyError failed on geitem/loc lookups (#31867)

mroeschke committed:
DOC: Reorder 1.0 releases in whatsnew/index.rst (#32309)

sathyz committed:
validate parse_dates - added nested list case

sathyz committed:
format code based on ci checks

sathyz committed:
Merge remote-tracking branch 'origin/issue-31251-1' into issue-31251-1

jreback commented:
closing as duplicate of #32320
